appengine/genetics/js/blocks.js

```diff
@@ -0,0 +1,445 @@
+        'AGGRESSIVENESS': BlocklyGames.getMsg('Genetics_aggressivenessTooltip'),
+        'FERTILITY': BlocklyGames.getMsg('Genetics_fertilityTooltip'),
+        'START_FERTILITY':
+            BlocklyGames.getMsg('Genetics_startFertilityTooltip'),
+        'SEX': BlocklyGames.getMsg('Genetics_sexTooltip'),
+        'AGE': BlocklyGames.getMsg('Genetics_ageTooltip'),
+        'ID': BlocklyGames.getMsg('Genetics_idTooltip'),
+        'PICK_FIGHT': BlocklyGames.getMsg('Genetics_pickFightOwnerTooltip'),
+        'PROPOSE_MATE': BlocklyGames.getMsg('Genetics_proposeMateOwnerTooltip'),
+        'ACCEPT_MATE': BlocklyGames.getMsg('Genetics_acceptMateOwnerTooltip')
+      };
+      return TOOLTIPS[mode];
+    });
+  },
+  /**
+   * Modify this block to have the correct output type.
+   * @param {string} newProp Either a property requesting a function owner that
+   * returns a string or a property that returns a number.
+   * @private
+   * @this Blockly.Block
+   */
+  updateType_: function(newProp) {
+    if (newProp == 'SEX') {
+      this.outputConnection.setCheck('String');
+    } else {
+      this.outputConnection.setCheck('Number');
+    }
+  }
+};
+
+/**
+ * Defines the JavaScript generation for getProperties.
+ * @param {Blockly.Block} block
+ * @return {!Array.<string|number>}
+ */
+Blockly.JavaScript['genetics_getProperties'] = function(block) {
+  // Generate JavaScript for getting mouse property.
+  var mouse = Blockly.JavaScript.valueToCode(block, 'MOUSE',
+          Blockly.JavaScript.ORDER_NONE) || 'me()';
+  var property = block.getFieldValue('PROPERTY');
+  var code = mouse + '.';
+  switch (property) {
+    case 'SIZE':
+      code += 'size';
+      break;
+    case 'AGGRESSIVENESS':
+      code += 'aggressiveness';
+      break;
+    case 'START_AGGRESSIVENESS':
+      code += 'startAggressiveness';
+      break;
+    case 'FERTILITY':
+      code += 'fertility';
+      break;
+    case 'START_FERTILITY':
+      code += 'startFertility';
+      break;
+    case 'SEX':
+      code += 'sex';
+      break;
+    case 'AGE':
+      code += 'age';
+      break;
+    case 'ID':
+      code += 'id';
+      break;
+    case 'PICK_FIGHT':
+      code += 'pickFightOwner';
+      break;
+    case 'PROPOSE_MATE':
+      code += 'proposeMateOwner';
+      break;
+    case 'ACCEPT_MATE':
+      code += 'acceptMateOwner';
+      break;
+    default:
+      throw 'Unknown mouse property: ' + property;
+  }
+  return [code, Blockly.JavaScript.ORDER_MEMBER];
+};
+
+/**
+ * Block that defines constants for possible sexes of a mouse.
+ */
+Blockly.Blocks['genetics_sex'] = {
+  /**
+   * Initializes the sex block.
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": "Sex.%1",
+      "args0": [
+        {
+          "type": "field_dropdown",
+          "name": "TYPE",
+          "options": [
+            ['Male', 'MALE'],
+            ['Female', 'FEMALE']
```

Comment:
it seems like the XML properties are all uppercase so that's what I followed this convention. The implementation of "pond_math_single" is quite similar with pairs like Math.abs and ABS.